kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/InBandKeyManager.java

```diff
@@ -194,7 +200,7 @@ private CompletableFuture<Void> encrypt(@NonNull EncryptionScheme<K> encryptionS
     private void rotateKeyContext(@NonNull EncryptionScheme<K> encryptionScheme, KeyContext keyContext) {
         keyContext.destroy();
         K kekId = encryptionScheme.kekId();
-        keyContextCache.put(kekId, makeKeyContext(kekId).get());
+        keyContextCache.synchronous().invalidate(kekId);
```

Comment:
After one of the other comments I expected to see more happening here than just the call to invalidate. Am I missing something? Or is the work the listeners were doing postponed to a future PR?

Reply:
https://github.com/kroxylicious/kroxylicious/pull/788#discussion_r1418048814

I found that the listeners were fired async on the forkjoinpool, but the current mutex type code depends on destroy happening while something is locked on the KeyContext

Reply:
The locking/mutex architecture is very up for debate/evolution in any case.